src/nvim/api/buffer.c

```diff
@@ -86,6 +86,12 @@ void buffer_del_line(Buffer buffer, Integer index, Error *err)
   FUNC_ATTR_DEFERRED
 {
   Array array = ARRAY_DICT_INIT;
+  buf_T *buf = find_buffer_by_handle(buffer, err);
+
+  if (index >= buf->b_ml.ml_line_count) {
+    return;
+  }
+
```

Comment:
Ah, okay. Updating the snippet with `inbounds()` sounds good, but also check for `buf != NULL`.